stablehlo/transforms/TypeConversion.h

```diff
@@ -28,99 +33,189 @@ namespace vhlo {
-        auto convertedEncoding = convertEncoding(encoding);
-        if (!convertedEncoding) return {};
-        return RankedTensorType::get(type.getShape(), type.getElementType(),
-                                     convertedEncoding);
-      }
-      return type;
-    });
-  };
+  VersionedTypeConverterBase() : TypeConverter(){};
 
   virtual ~VersionedTypeConverterBase() = default;
 
-  // Checks whether the given dialect is the source dialect of the type
-  // conversion (e.g. StableHLO for StablehloToVhloTypeConverter).
-  virtual bool isSourceDialect(Dialect& dialect) = 0;
-
   virtual Attribute convertEncoding(Attribute attr) = 0;
 };
 
 class StablehloToVhloTypeConverter : public VersionedTypeConverterBase {
  public:
   StablehloToVhloTypeConverter() : VersionedTypeConverterBase() {
+    addConversion([](Type type) -> Type {
+      if (type.getDialect().getNamespace() == "vhlo") {
+        return type;
+      }
+      LLVM_DEBUG(llvm::dbgs() << "Invalid type: " << type << '\n');
+      return Type();
+    });
     addConversion([](stablehlo::TokenType token) -> Type {
-      LLVM_DEBUG(llvm::dbgs() << "Converting TokenType\n");
       return TokenType::get(token.getContext());
     });
+
+    // Forked Types
+    addConversion([&](BFloat16Type type) {
+      return BFloat16V1Type::get(type.getContext());
+    });
+    addConversion([&](ComplexType type) {
+      return ComplexV1Type::get(type.getContext(),
+                                convertType(type.getElementType()));
+    });
+    addConversion([&](Float16Type type) {
+      return Float16V1Type::get(type.getContext());
+    });
+    addConversion([&](Float32Type type) {
+      return Float32V1Type::get(type.getContext());
+    });
+    addConversion([&](Float64Type type) {
+      return Float64V1Type::get(type.getContext());
+    });
+    addConversion(
+        [&](IndexType type) { return IndexV1Type::get(type.getContext()); });
+    addConversion([&](IntegerType type) {
+      return IntegerV1Type::get(type.getContext(), type);
+    });
+    addConversion([&](RankedTensorType type) -> Type {
+      auto encoding = type.getEncoding();
+      auto convertedEncoding = encoding ? convertEncoding(encoding) : encoding;
+      auto convertedElementType = convertType(type.getElementType());
+      if ((encoding && !convertedEncoding) || !convertedElementType) return {};
+      return RankedTensorV1Type::get(type.getContext(), type.getShape(),
+                                     convertedElementType, convertedEncoding);
+    });
+    addConversion([&](TupleType type) -> Type {
+      SmallVector<Type> convertedTypes;
+      if (failed(convertTypes(type.getTypes(), convertedTypes))) return {};
+      return vhlo::TupleV1Type::get(type.getContext(), convertedTypes);
+    });
+    addConversion([&](quant::UniformQuantizedType type) -> Type {
+      Type storage = convertType(type.getStorageType());
+      Type expressed = convertType(type.getExpressedType());
```

Comment:
`convertedStorageType` and `convertedExpressedType`? (Same below).

Reply:
I don't think this has been addressed.

Reply:
Fixed it in the StableHLO->VHLO conversion, missed it in the VHLO->StableHLO. Fixed in both now.